consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+                'New fork head {} is not a PoET block',
+                str(cm.exception))
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_cur_fork_head_not_valid_block(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ Test verifies that if the current fork head is not a valid block,
+            and if new_fork_head.previous_block_id == cur_fork_head.identifier
+            then the new fork head switches consensus. Otherwise, exits -
+            trying to compare a PoET block {} to a non-PoET block that is not
+            the direct predecessor
+        """
+
+        # create a mock_validator_registry_view
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # create a mock_wait_certificate that does nothing in check_valid
+        mock_wait_certificate = mock.Mock()
+        mock_wait_certificate.check_valid.return_value = None
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head that is used in
+        # poet_fork_resolver.compare()
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [None,
+             mock_wait_certificate,
+             None,
+             mock_wait_certificate]
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_poet_enclave_module
+        mock_poet_enclave_module = mock.Mock()
+        mock_poet_enclave_module.return_value = \
+            mock_poet_enclave_factory.get_poet_enclave_module.return_value
```

Comment:
Same comment about the mock poet enclave factory as above - you really mean to return `mock_poet_enclave_module` as the result of `mock_poet_enclave_factor.get_poet_enclave_module.return_value` , although if the code doesn't really do anything with the enclave besides pass it to other methods, then you can probably not bother setting `.return_value` as a `mock.Mock` object is probably sufficient.